### ISSUE TITLE
macOS builds failing

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#4369]
## Description

Could be related to the recent NumPy 1.18 release, or just an update to Travis compilers, not sure. Here's an example failure from #4356:

https://travis-ci.org/scikit-image/scikit-image/jobs/628839752#L2728-L2730

```
  g++: skimage/feature/_cascade.cpp
2729  error: invalid argument '-std=c99' not allowed with 'C++'
2730  error: Command "g++ -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -std=c99 -I/Users/travis/venv/lib/python3.6/site-packages/numpy/core/include -I/Users/travis/venv/lib/python3.6/site-packages/numpy/core/include -I/Library/Frameworks/Python.framework/Versions/3.6/include/python3.6m -c skimage/feature/_cascade.cpp -o build/temp.macosx-10.9-x86_64-3.6/skimage/feature/_cascade.o -MMD -MF build/temp.macosx-10.9-x86_64-3.6/skimage/feature/_cascade.o.d" failed with exit status 1
2731
```

**Comment by hmaarrfk** [comment]
I don't understand where that compilation flag is being added. It really isn't obvious where the fix should go. In our scripts, we never specify c99

**Comment by hmaarrfk** [comment]
It seems related to this new feature
https://github.com/numpy/numpy/blob/4f972297277e09f41f315fd75510cb313926fe67/doc/source/release/1.18.0-notes.rst#stdc99-added-if-compiler-is-named-gcc

**Comment by hmaarrfk** [comment]
https://github.com/numpy/numpy/blob/master/numpy/distutils/ccompiler.py#L538

**Comment by hmaarrfk** [comment]
@mattip any clue how to get around this? I guess we were using gcc on OSX, and the build flags weren't really happy.

**Comment by mattip** [comment]
The check specifically checks for `gcc`, how is it being changed to `g++`?

**Comment by mattip** [comment]
Should be fixed in numpy/numpy#15194
